ITechArt.FlightBookingsAPI.Services/Services/UsersService.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+using ITechArt.FlightBookingsAPI.Domain.Errors;
+using ITechArt.FlightBookingsAPI.Domain.Models;
+using ITechArt.FlightBookingsAPI.Services.Interfaces;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.IdentityModel.Tokens;
+
+namespace ITechArt.FlightBookingsAPI.Services.Services;
+
+public class UsersService : IUsersService
+{
+    private const string UserRole = "User";
+    private const string AdminRole = "Admin";
+
+    private readonly UserManager<User> _userManager;
+
+    public UsersService(UserManager<User> userManager)
+    {
+        _userManager = userManager;
+    }
+    
+    public async Task Register(User user, string password)
+    {
+        var result = await _userManager.CreateAsync(user, password);
+        if (!result.Succeeded)
+        {
+            throw new BadRequestException(result.Errors.First().Description);
```

Comment:
I don't think there's a real reason to use BadRequestException here as it is recall of Bad Request HTTP-code that's basically from another layer. I'd prefer to use here ArgumentException instead of BadRequestException.